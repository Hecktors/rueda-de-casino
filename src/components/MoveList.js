import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

MoveList.propTypes = { moves: PropTypes.array.isRequired }

export default function MoveList({ moves }) {
  if (!Array.isArray(moves)) return <div className="error-msg">Data Error</div>

  const listItems = moves.map(({ name, id }) => <li key={id}>{name}</li>)

  const content =
    listItems.length > 0 ? (
      listItems
    ) : (
      <span className="warning">No move selected</span>
    )

  return <ListStyled>{content}</ListStyled>
}

const ListStyled = styled.ul`
  width: 100%;
  padding-left: 50px;
  li {
    margin: 5px;
    list-style: none;
  }

  & .warning {
    color: orange;
  }
`
