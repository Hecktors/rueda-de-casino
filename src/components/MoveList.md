```jsx
import moves from '../mockdata/moves.json'
;<div>
  <MoveList moves={moves} isPaused={true} />
  <MoveList moves={moves} isPaused={false} />
</div>
```
