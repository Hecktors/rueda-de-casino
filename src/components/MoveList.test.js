import MoveList from './MoveList'
import { render } from '@testing-library/react'
const testMoves = [
  {
    id: 1,
    name: 'la prima',
  },
  {
    id: 2,
    name: 'sombrero',
  },
]

describe('MoveList', () => {
  it('renders MoveList', () => {
    const { container } = render(<MoveList moves={testMoves} />)
    expect(container).toMatchSnapshot()
  })

  it('shows move names', () => {
    const { getByText } = render(<MoveList moves={testMoves} />)
    expect(getByText('la prima')).toBeInTheDocument()
    expect(getByText('sombrero')).toBeInTheDocument()
  })
})
