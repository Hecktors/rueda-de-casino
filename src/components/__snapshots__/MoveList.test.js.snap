// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoveList renders MoveList 1`] = `
<div>
  <div
    class="MoveList__MoveListStyled-skwchu-0 dhfzQJ"
  >
    <ul>
      <li>
        <button
          data-testid="listitem-button"
        >
          la prima
          <span />
        </button>
      </li>
      <li>
        <button
          data-testid="listitem-button"
        >
          sombrero
          <span />
        </button>
      </li>
    </ul>
  </div>
</div>
`;
